### ISSUE TITLE
Warn about unused macros

### ISSUE THREAD

**Issue by kmcallister** [rust-lang/rust#15128]

**Comment by steveklabnik** [comment]
Triage: 

```
macro_rules! foo {
    (x => $e:expr) => (println!("mode X: {}", $e));
    (y => $e:expr) => (println!("mode Y: {}", $e));
}

fn main() {
    foo!(y => 3);
}
```

commenting out the `foo!` in `main` does not warn about never using `foo!`.

**Comment by steveklabnik** [comment]
Since new lints have a big impact on users of rustc, the policy is that they should go through the RFC process like other user-facing changes. As such, I'm going to give this one a close, but if anyone comes across this ticket and wants this lint, consider adding it to clippy and/or writing up an RFC. Thanks!
